### PR TITLE
Add Gauge export support in Prometheus

### PR DESCRIPTION
**Pull request by cijothomas**
Fixes #.

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs [hunk 1]
```diff
@@ -93,6 +106,21 @@ private static void WriteSum(StreamWriter writer, PrometheusMetricBuilder builde
             builder.Write(writer);
         }
 
+        private static void WriteGauge(StreamWriter writer, PrometheusMetricBuilder builder, IEnumerable<KeyValuePair<string, object>> labels, double doubleValue)
```
Comment (utpilla) [inline src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs:109]: Does this method only take a `double` value? Do we need a method that accepts `long` as well?

Comment (cijothomas) [reply src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs:109]: Prometheus exposition format seem to be only accepting float/double, so casting long to double for now.

File: src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs [hunk 2]
```diff
@@ -59,6 +60,18 @@ public static void WriteMetricsCollection(this PrometheusExporter exporter, Stre
...
+                    else if (metric.MetricType == MetricType.DoubleGauge)
+                    {
+                        var gaugeMetric = metric as IGaugeMetric;
+                        var doubleValue = (double)gaugeMetric.LastValue.Value;
+                        WriteGauge(writer, builder, metric.Attributes, doubleValue);
+                    }
+                    else if (metric.MetricType == MetricType.LongGauge)
+                    {
+                        var gaugeMetric = metric as IGaugeMetric;
+                        var longValue = (long)gaugeMetric.LastValue.Value;
```
Comment (utpilla) [inline src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs:72]: Should we skip this for now then? We could just have the same logic as MetricType.DoubleGauge.

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs (hunk 2 above)
Comment (cijothomas) [reply src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs:72]: I'll mark a TODO here to comeback to this, in the my next PR (which adds histogram)

**Pull request merged**